src/Testcontainers.Qdrant/QdrantBuilder.cs

```diff
@@ -0,0 +1,77 @@
+namespace Testcontainers.Qdrant;
+
+/// <inheritdoc cref="ContainerBuilder{TBuilderEntity, TContainerEntity, TConfigurationEntity}" />
+[PublicAPI]
+public sealed class QdrantBuilder : ContainerBuilder<QdrantBuilder, QdrantContainer, QdrantConfiguration>
+{
+	public const string QdrantImage = "qdrant/qdrant:v1.8.3";
+
+	public const ushort QdrantHttpPort = 6333;
+
+	public const ushort QdrantGrpcPort = 6334;
+	
+	public const string QdrantTlsCertFilePath = "/qdrant/tls/cert.pem";
+	
+	public const string QdrantTlsKeyFilePath = "/qdrant/tls/key.pem";
+
+	public QdrantBuilder() : this(new QdrantConfiguration()) =>
+		DockerResourceConfiguration = Init().DockerResourceConfiguration;
+
+	private QdrantBuilder(QdrantConfiguration dockerResourceConfiguration) : base(dockerResourceConfiguration) =>
+		DockerResourceConfiguration = dockerResourceConfiguration;
+	
+	/// <summary>
+	/// The API key used to secure the instance. A certificate and private key should also be
+	/// provided to <see cref="WithCertificate"/> to enable Transport Layer Security (TLS).
+	/// </summary>
+	/// <param name="apiKey">The API key</param>
+	public QdrantBuilder WithApiKey(string apiKey) =>
+		Merge(DockerResourceConfiguration, new QdrantConfiguration(apiKey: apiKey))
+			.WithEnvironment("QDRANT__SERVICE__API_KEY", apiKey);
+
+	/// <summary>
+	/// A certificate and private key to enable Transport Layer Security (TLS).
+	/// </summary>
+	/// <param name="certificate">A public certificate in PEM format</param>
+	/// <param name="privateKey">A private key for the certificate in PEM format</param>
+	public QdrantBuilder WithCertificate(string certificate, string privateKey)
+	{
+		return Merge(DockerResourceConfiguration, new QdrantConfiguration(certificate: certificate, privateKey: privateKey))
+			.WithEnvironment("QDRANT__SERVICE__ENABLE_TLS", "1")
+			.WithResourceMapping(Encoding.UTF8.GetBytes(certificate), QdrantTlsCertFilePath)
+			.WithEnvironment("QDRANT__TLS__CERT", QdrantTlsCertFilePath)
+			.WithResourceMapping(Encoding.UTF8.GetBytes(privateKey), QdrantTlsKeyFilePath)
+			.WithEnvironment("QDRANT__TLS__KEY", QdrantTlsKeyFilePath);
+	}
+    
+	/// <inheritdoc />
+	public override QdrantContainer Build()
+	{
+		Validate();
+		return new QdrantContainer(DockerResourceConfiguration);
+	}
+
+	/// <inheritdoc />
+	protected override QdrantBuilder Init() =>
+		base.Init()
+			.WithImage(QdrantImage)
+			.WithPortBinding(QdrantHttpPort, true)
+			.WithPortBinding(QdrantGrpcPort, true)
+			.WithWaitStrategy(Wait.ForUnixContainer()
+				.UntilMessageIsLogged(".*Actix runtime found; starting in Actix runtime.*"));
```

Comment:
shouldn't it wait for `/readyz` in port `6333`?

Reply:
I've changed the wait for strategy to use the readyz endpoint instead, handling the case where a certificate is provided. Ability to provide a certificate has to be supported for net462, because of the underlying gRPC packages used in net462- see [qdrant .NET SDK README](https://github.com/qdrant/qdrant-dotnet).